api/adaptive_load/adaptive_load.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "api/adaptive_load/benchmark_result.proto";
+import "api/adaptive_load/metrics_plugin.proto";
+import "api/adaptive_load/metric_spec.proto";
+import "api/adaptive_load/step_controller.proto";
+import "api/client/options.proto";
+import "google/protobuf/duration.proto";
+import "google/rpc/status.proto";
+import "validate/validate.proto";
+
+// Parameters describing the adjusting and testing stages of an adaptive load
+// session, which consists of a series of Nighthawk benchmarks probing for
+// the optimal load on the system, followed by a longer benchmark to validate
+// the values. Load adjustments are calculated by the selected StepController
+// plugin. Metrics can come from Nighthawk stats and counters or
+// platform-specific data sources via MetricsPlugins.
+message AdaptiveLoadSessionSpec {
+  // Settings for MetricsPlugins that obtain metrics from outside sources.
+  // An entry is required for every plugin referred to by metric_thresholds,
+  // other than the "builtin" plugin.
+  repeated MetricsPluginConfig metrics_plugin_configs = 1;
+  // Metrics and thresholds that determine load adjustments.
+  repeated MetricSpecWithThreshold metric_thresholds = 2 [(validate.rules).repeated .min_items = 1];
+  // Metrics that are collected and included in the output but not taken into
+  // account when adjusting the load.
+  repeated MetricSpec informational_metric_specs = 3;
+  // A proto describing Nighthawk Service traffic. See
+  // https://github.com/envoyproxy/nighthawk/blob/master/api/client/options.proto
+  //
+  // The adaptive load controller will return an error if the |duration| or
+  // |open_loop| fields are set within |nighthawk_traffic_options|. The
+  // controller will also be configured to overwrite at least one of the
+  // numerical fields during the search, such as requests_per_second, so any
+  // value of those fields specified here will be ignored.
+  //
+  // All other fields in |nighthawk_traffic_options| are passed through to the
+  // Nighthawk Service.
+  //
+  // Note that |concurrency| in |nighthawk_traffic_options| multiplies the total
+  // RPS sent.
+  nighthawk.client.CommandLineOptions nighthawk_traffic_template = 4;
+  // The duration of each short benchmark during the adjusting stage. Must be
+  // set. Reasonable value: 10 seconds.
```

Comment:
Should we set the reasonable value as the default instead of commenting it?

Applies to multiple locations with the goal of "out of the box" usability.

Reply:
Done.